src/node/blockstorage.cpp

```diff
@@ -387,7 +387,13 @@ bool BlockManager::LoadBlockIndex(const std::optional<uint256>& snapshot_blockha
     }
 
     if (snapshot_blockhash) {
-        const AssumeutxoData au_data = *Assert(GetParams().AssumeutxoForBlockhash(*snapshot_blockhash));
+        const auto& maybe_au_data = GetParams().AssumeutxoForBlockhash(*snapshot_blockhash);
+        if (!maybe_au_data) {
+            BlockValidationState state_dummy;
+            const std::string au_data_error = strprintf("Assumeutxo data not found for the given blockhash '%s'.", snapshot_blockhash->ToString());
+            return FatalError(m_opts.notifications, state_dummy, au_data_error, _(au_data_error.c_str()));
```

Comment:
In commit "assumeutxo, blockstorage: prevent core dump on invalid hash" (caae1c935e3ee8541fe065da43bfb2c3c8e57554)

Would be good to drop the `state_dummy` variable and  `_(au_data_error.c_str())` translation which doesn't really make sense, and just call m_opts.notifications->fatalError() instead of FatalError

Reply:
If i remove the translation string (the second parameter of the `fatalError()` function) the propagated error would be "`Error: A fatal internal error occured...`":

```
2023-10-25T02:29:57Z [init] Using obfuscation key for /home/pablo/.test_utxo_2/regtest/blocks/index: 0000000000000000
2023-10-25T02:29:57Z [init] *** Assumeutxo data not found for the given blockhash 'f09b5835f3f8b39481f2af3257bbc2e82845552d4d2d6d31cf520fc24263ed5b'.
2023-10-25T02:29:57Z [init] Error: A fatal internal error occurred, see debug.log for details
Error: A fatal internal error occurred, see debug.log for details
2023-10-25T02:29:57Z [init] Shutdown requested. Exiting.
2023-10-25T02:29:57Z [init] Shutdown: In progress...
2023-10-25T02:29:57Z [scheduler] scheduler thread exit
2023-10-25T02:29:57Z [shutoff] Flushed fee estimates to fee_estimates.dat.
2023-10-25T02:29:57Z [shutoff] Shutdown: done

```

So I have to update also the `expected_error` value in the new test `test_invalid_chainstate_scenarios` added by @theStack.

Reply:
Updated it.

Reply:
Ah... I'd need to do what @theStack [mentioned below](https://github.com/bitcoin/bitcoin/pull/28698#pullrequestreview-1698178157).